Drivers/seds/UartHwas/UartHwas.xml

```diff
@@ -0,0 +1,1330 @@
+                          <Condition>
+                            <Condition>
+                              <FirstOperand variableRef="config.mInstance" />
+                              <ComparisonOperator>equals</ComparisonOperator>
+                              <Value value="UartHwas_Instance_2" />
+                            </Condition>
+                          </Condition>
+                          <OnConditionTrue>
+                            <Assignment outputVariableRef="uartVar2">
+                              <VariableRef variableRef="uart" />
+                            </Assignment>
+                          </OnConditionTrue>
+                          <OnConditionFalse>
+                            <Conditional>
+                              <LongDescription>Is instance equal to 3?</LongDescription>
+                              <Condition>
+                                <Condition>
+                                  <FirstOperand variableRef="config.mInstance" />
+                                  <ComparisonOperator>equals</ComparisonOperator>
+                                  <Value value="UartHwas_Instance_3" />
+                                </Condition>
+                              </Condition>
+                              <OnConditionTrue>
+                                <Assignment outputVariableRef="uartVar3">
+                                  <VariableRef variableRef="uart" />
+                                </Assignment>
+                              </OnConditionTrue>
+                              <OnConditionFalse>
+                                <Conditional>
+                                  <LongDescription>Is instance equal to 4?</LongDescription>
+                                  <Condition>
+                                    <Condition>
+                                      <FirstOperand variableRef="config.mInstance" />
+                                      <ComparisonOperator>equals</ComparisonOperator>
+                                      <Value value="UartHwas_Instance_4" />
+                                    </Condition>
+                                  </Condition>
+                                  <OnConditionTrue>
+                                    <Assignment outputVariableRef="uartVar4">
+                                      <VariableRef variableRef="uart" />
+                                    </Assignment>
+                                  </OnConditionTrue>
+                                </Conditional>
+                              </OnConditionFalse>
+                            </Conditional>
+                          </OnConditionFalse>
+                        </Conditional>
+                      </OnConditionFalse>
+                    </Conditional>
+                  </OnConditionFalse>
+                </Conditional>
+              </Body>
+            </Activity>
+            <Activity name="GetLocalUart">

```

Comment:
Let's discuss how this works

Reply:
It works in the same way as https://github.com/n7space/TASTE-SAMV71-Runtime/blob/main/src/Hal/Hal.c.
The interrupt handler needs the reference to the UART that triggered that interrupt. It needs it as it must read the values from a couple of registers. I have added the UART variables, an equivalent of the static uartXHandles from the example above, to pass correct register addresses to the interrupt handler.